secretary.py

```diff
@@ -306,20 +306,30 @@ def _prepare_template_tags(self, xml_document):
             parent.removeChild(discard)
 
 
-    def _unescape_entities(self, xml_text):
-        # unescape XML entities gt and lt
+    @staticmethod
+    def _unescape_entities(xml_text):
+        """
+        Strips tags of the form <text:span ...> from inside Jinja elements
+        and unescapes HTML codes for >, <, & and "
+        """
         unescape_rules = {
+            r'(?is)({[{|%].*)(<.?text:s.*?>)(.*[%|}]})': r'\1\3',
+
             r'(?is)({[{|%].*)(&gt;)(.*[%|}]})': r'\1>\3',
             r'(?is)({[{|%].*)(&lt;)(.*[%|}]})': r'\1<\3',
-            r'(?is)({[{|%].*)(<.?text:s.?>)(.*[%|}]})': r'\1 \3',
+            r'(?is)({[{|%].*)(&amp;)(.*[%|}]})': r'\1&\3',
+            r'(?is)({[{|%].*)(&quot;)(.*[%|}]})': r'\1"\3'
         }
 
         for p, r in unescape_rules.items():
-            xml_text = re.sub(p, r, xml_text)
+            count = None
+            while count != 0:
+                xml_text, count = re.subn(p, r, xml_text)
 
         return xml_text
 
-    def _encode_escape_chars(self, xml_text):
+    @staticmethod
```

Comment:
I figured these two could be made into static methods since they do not rely on `self`.